src/mono/wasm/runtime/library_mono.js

```diff
@@ -1445,9 +1445,9 @@ var MonoSupportLib = {
 			if (options == null)
 				options = {}
 			if (!('write_at' in options))
-				options.write_at = 'WebAssembly.Runtime::StopProfile';
+				options.write_at = 'Interop/Runtime::StopProfile';
```

Comment:
If you change change this value you shouldn't need to load all the interop stuff.  You can point it at a function in the sample

Reply:
To call `DumpAOTProfile` we would still have to load the interop stuff.

Reply:
The code is there already if the System.Private.Runtime.InteropServices.JavaScript assembly is being loaded (and DumpProfile hasn't been linked out)

Reply:
and by change the value I mean set it when generating the options.  The situation is kind of a mess because that assembly isn't public right now but we can deal with it differently until that changes.

Reply:
Ah, got it.